packages/json-rpc-provider/package.json

```diff
@@ -0,0 +1,27 @@
+{
+  "name": "@near-js/json-rpc-provider",
+  "version": "0.0.1",
+  "description": "Client for communicating with the NEAR RPC using JSON-RPC",
+  "main": "lib/index.js",
+  "scripts": {
+    "build": "pnpm compile",
+    "compile": "tsc -p tsconfig.json"
```

Comment:
any ideas on testing here and core and naj? splitting them like here with the code would make sense to me

Reply:
Since the break was relatively clean I think it's just a matter of moving tests directly out of `near-api-js` but I haven't looked closely at the tests to verify. Some integration tests may end up staying in `near-api-js` if their dependencies are too thorny but I expect most can just live in their new packages.

However I'm first trying to work out a reasonable approach to CI dependencies with `pnpm` and `turborepo`. I may need to create NPM packages first if it ends up being a bootstrapping issue but in theory it should all work locally. Specifically `near-api-js` has an install-time dependency on `@near-js/core`, which must first have its own dependencies installed and be built (transpiled into JS). It seems like since the install and build steps are independent, they lack the awareness of the more complex relationship between dependencies.

Reply:
Fixed by removing the `prepare` command 🎉 

`prepare` was running the `build` command, which requires that `install` has already run... but `prepare` runs automatically when `install` is run, so every clean `build` fails because `install` that won't run until `build` succeeds. By removing `prepare`, `pnpm` can do its thing and manage workspace dependencies.

Reply:
I've moved the low hanging fruit, some of the others will require a more thoughtful refactor.